File: 0649_dota2Senate/main.py
from collections import deque

class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        radiant, dire = deque(), deque()
        senateLen = len(senate)
        for idx, c in enumerate(senate):
            if c == 'R':
                radiant.append(idx)
            else:
                dire.append(idx)
        
        while radiant and dire:
            radiantFront = radiant.popleft()
            direFront = dire.popleft()
            if radiantFront < direFront:
                radiant.append(radiantFront + senateLen)
            else:
                dire.append(direFront + senateLen)
        
        return "Radiant" if radiant else "Dire"

File: 0649_dota2Senate/main.cpp
#include <string>
#include <queue>

class Solution {
public:
    std::string predictPartyVictory(std::string senate) {
        std::queue<int> radiant;
        std::queue<int> dire;

        int len = senate.size();
        for(int i = 0; i < len; ++i) {
            if(senate[i] == 'R')
                radiant.push(i);
            else
                dire.push(i);
        }

        int radiantFront, direFront; 
        while(!radiant.empty() && !dire.empty()) {
            radiantFront = radiant.front();
            direFront = dire.front();
            radiant.pop();
            dire.pop();
            if(radiantFront < direFront)
                radiant.push(radiantFront + len);
            else
                dire.push(direFront + len);
        }

        return !radiant.empty() ? "Radiant" : "Dire";
    }
};